# **Learning Pandas**

Pandas is a powerful library. Let's just see what it can does to know its importance.
The external dataset to be downloaded: https://www.dropbox.com/scl/fi/0uiujtei423te1q4kvrny/diabetes.csv?rlkey=20xvytca6xbio4vsowi2hdj8e&e=1&dl=0

Download and upload to Google Colab with name diabetes.csv

In [21]:
#importing
import pandas as pd
import numpy as np

# **DataFrame Creation Method 1**

DataFrame is a "**two-dimensional**", "**tabular**" data with "**labeled**" axis (rows and columns)

In [22]:
#Creating a DataFrame
#from sklearn.datasets import load_boston
#b_dt= load_boston()

The boston dataset is removed from the library in the version upgrade for ethical issues. A different dataset was referred. Whooh. Let's use it!

In [23]:
from sklearn.datasets import fetch_california_housing
housing = fetch_california_housing()

In [24]:
type(housing)

sklearn.utils._bunch.Bunch

In [25]:
print(housing)

{'data': array([[   8.3252    ,   41.        ,    6.98412698, ...,    2.55555556,
          37.88      , -122.23      ],
       [   8.3014    ,   21.        ,    6.23813708, ...,    2.10984183,
          37.86      , -122.22      ],
       [   7.2574    ,   52.        ,    8.28813559, ...,    2.80225989,
          37.85      , -122.24      ],
       ...,
       [   1.7       ,   17.        ,    5.20554273, ...,    2.3256351 ,
          39.43      , -121.22      ],
       [   1.8672    ,   18.        ,    5.32951289, ...,    2.12320917,
          39.43      , -121.32      ],
       [   2.3886    ,   16.        ,    5.25471698, ...,    2.61698113,
          39.37      , -121.24      ]]), 'target': array([4.526, 3.585, 3.521, ..., 0.923, 0.847, 0.894]), 'frame': None, 'target_names': ['MedHouseVal'], 'feature_names': ['MedInc', 'HouseAge', 'AveRooms', 'AveBedrms', 'Population', 'AveOccup', 'Latitude', 'Longitude'], 'DESCR': '.. _california_housing_dataset:\n\nCalifornia Housing dataset\n-

Yes, that seems like a dictionary, and thus bunch is Dictionary type object! Noice.

In [26]:
housing_df=pd.DataFrame(housing.data)

In [27]:
housing_df.head()

0     1         2         3       4         5      6       7
0  8.3252  41.0  6.984127  1.023810   322.0  2.555556  37.88 -122.23
1  8.3014  21.0  6.238137  0.971880  2401.0  2.109842  37.86 -122.22
2  7.2574  52.0  8.288136  1.073446   496.0  2.802260  37.85 -122.24
3  5.6431  52.0  5.817352  1.073059   558.0  2.547945  37.85 -122.25
4  3.8462  52.0  6.281853  1.081081   565.0  2.181467  37.85 -122.25

In [28]:
housing_df2=pd.DataFrame(housing.data, columns= housing.feature_names)
housing_df2.head()


MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  Latitude  \
0  8.3252      41.0  6.984127   1.023810       322.0  2.555556     37.88   
1  8.3014      21.0  6.238137   0.971880      2401.0  2.109842     37.86   
2  7.2574      52.0  8.288136   1.073446       496.0  2.802260     37.85   
3  5.6431      52.0  5.817352   1.073059       558.0  2.547945     37.85   
4  3.8462      52.0  6.281853   1.081081       565.0  2.181467     37.85   

   Longitude  
0    -122.23  
1    -122.22  
2    -122.24  
3    -122.25  
4    -122.25

The addition of columns=housing.feature_names has added the column names nicely in the dataset! The feature_names were given in the dictionary. So this was a nice use of extracting info from the dictionary type objects and the powerfulness of dictionary objects!

In [29]:
housing_df2.shape

(20640, 8)

shape testing done. There are 20,640 rows and 8 columns!

# **Method 2**
This is a DataFrame creation method where csv files will be used directly from the files section, rather than dictionary files. CSV files are basically data contained as a Comma Separated Values file.

In [30]:
dia_df=pd.read_csv('/content/diabetes.csv')
print(dia_df)

     Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                  

In [31]:
type(dia_df)

pandas.core.frame.DataFrame

In [32]:
dia_df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [33]:
dia_df.shape

(768, 9)

## **DataFrame to csv conversion**

In [34]:
dia_df.to_csv('diabetes_df.csv')

In [35]:
housing_df2.to_csv('house.csv')

You can find these under the files section of you google colaboratory! Downloading them is also straightforward. The exporting methods can easily by learned by googling about them.

# **Random value DataFrames**

So, using NumpY, we can create random numbers between 0 and 1, and then with some parameterization, we can use them to create a desired shaped DataFrame. Here are some examples:

In [36]:
rand_df1=pd.DataFrame(np.random.rand(20,8))

In [37]:
rand_df1.head()

0         1         2         3         4         5         6  \
0  0.377463  0.745756  0.043773  0.610861  0.572512  0.025987  0.991802   
1  0.124801  0.585201  0.623672  0.756853  0.623385  0.598924  0.560327   
2  0.961274  0.669149  0.669840  0.719108  0.970202  0.941338  0.702381   
3  0.772659  0.190259  0.096171  0.216042  0.372375  0.859790  0.677094   
4  0.088386  0.130117  0.438145  0.428476  0.966997  0.606975  0.696719   

          7  
0  0.739901  
1  0.904654  
2  0.300311  
3  0.778972  
4  0.952014

In [38]:
rand_df1.shape

(20, 8)

In [40]:
rand_df2=pd.DataFrame(np.random.rand(10,3))

In [41]:
rand_df2.head()

0         1         2
0  0.774346  0.759928  0.613820
1  0.836463  0.029379  0.081111
2  0.863606  0.391645  0.234402
3  0.876590  0.510894  0.486188
4  0.716474  0.878516  0.209904

In [42]:
rand_df2.shape

(10, 3)

#**Inspecting Data**

When you find a data, you have to inspect it at first, kinda laking looking around to get some basic informations like shape, element number, whether there is a missing or null value, the missing values' locations, and many more. Let's start!

In [47]:
print(housing_df.shape)
print(housing_df.head())
print(housing_df.tail())
print(housing_df.info())

(20640, 8)
        0     1         2         3       4         5      6       7
0  8.3252  41.0  6.984127  1.023810   322.0  2.555556  37.88 -122.23
1  8.3014  21.0  6.238137  0.971880  2401.0  2.109842  37.86 -122.22
2  7.2574  52.0  8.288136  1.073446   496.0  2.802260  37.85 -122.24
3  5.6431  52.0  5.817352  1.073059   558.0  2.547945  37.85 -122.25
4  3.8462  52.0  6.281853  1.081081   565.0  2.181467  37.85 -122.25
            0     1         2         3       4         5      6       7
20635  1.5603  25.0  5.045455  1.133333   845.0  2.560606  39.48 -121.09
20636  2.5568  18.0  6.114035  1.315789   356.0  3.122807  39.49 -121.21
20637  1.7000  17.0  5.205543  1.120092  1007.0  2.325635  39.43 -121.22
20638  1.8672  18.0  5.329513  1.171920   741.0  2.123209  39.43 -121.32
20639  2.3886  16.0  5.254717  1.162264  1387.0  2.616981  39.37 -121.24
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 8 columns):
 #   Column  Non-Null Count  

The shape will give the row and column number, head and tail will give first 5 and last 5 rows, respectively, and info will give a general information. Here entries means total element number (20640 entries), non-null means not missing or existing values, dtype refers to data type.

##**Missing Value**
The missing value part is important, and you have to look out for missing values in any database while preparing it for any use. There are more ways to deal about it!

In [48]:
housing_df.isnull().sum()

0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    0
dtype: int64

In [50]:
dia_df.isnull().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

There are no missing values there it seems.



##**Couting similars**

There is a value_count method which can be used to know about the appearance frequency of any numbers. For example, let's say you want to solve a murder mystery. Then you will look for frequencies of the numbers that called the victim. This is something similar! Let's look in the diabetes dataset.

In [53]:
dia_df.head()


Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0

In [54]:
dia_df.tail()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0

From them, Pregnancy, Outcome, Age might be good to check for the purpose of checking data bias.

In [63]:
dia_df.value_counts(['Outcome'])

Outcome
0          500
1          268
Name: count, dtype: int64

In [64]:
dia_df.value_counts(['Pregnancies'])

Pregnancies
1              135
0              111
2              103
3               75
4               68
5               57
6               50
7               45
8               38
9               28
10              24
11              11
13              10
12               9
14               2
15               1
17               1
Name: count, dtype: int64

In [66]:
dia_df.value_counts(['Age'])

Age
22     72
21     63
25     48
24     46
23     38
28     35
26     33
27     32
29     29
31     24
41     22
30     21
37     19
42     18
33     17
38     16
36     16
32     16
45     15
34     14
40     13
43     13
46     13
39     12
35     10
52      8
44      8
50      8
51      8
58      7
54      6
47      6
53      5
60      5
49      5
57      5
48      5
66      4
62      4
63      4
55      4
59      3
56      3
65      3
67      3
61      2
69      2
64      1
68      1
70      1
72      1
81      1
Name: count, dtype: int64

##**Mean function along with grouping!**

This is a nice one. You will know its power by seeing it

In [67]:
dia_df.groupby(['Outcome']).mean()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
Outcome                                                                      
0           3.298000  109.980000      68.184000      19.664000   68.792000   
1           4.865672  141.257463      70.824627      22.164179  100.335821   

               BMI  DiabetesPedigreeFunction        Age  
Outcome                                                  
0        30.304200                  0.429734  31.190000  
1        35.142537                  0.550500  37.067164

In [78]:
dia_df.groupby(['Pregnancies']).mean()

Glucose  BloodPressure  SkinThickness     Insulin        BMI  \
Pregnancies                                                                    
0            123.000000      67.153153      22.270270   81.675676  34.290090   
1            112.748148      67.792593      24.437037   98.674074  31.372593   
2            110.796117      63.252427      21.601942   85.844660  30.583495   
3            123.586667      66.586667      20.080000   87.453333  30.425333   
4            125.117647      70.029412      15.882353   69.441176  32.141176   
5            118.859649      76.210526      17.385965   57.298246  33.192982   
6            120.800000      68.420000      17.640000   63.580000  30.290000   
7            136.444444      70.777778      20.288889   84.466667  32.631111   
8            131.736842      75.184211      17.315789   92.815789  31.568421   
9            131.392857      77.892857      20.892857   62.428571  31.707143   
10           120.916667      70.208333      15.708333   34.791667  30.641667   
11           126.454545      74.181818      21.727273   65.454545  38.563636   
12           113.555556      76.333333      27.111111  112.555556  32.344444   
13           125.500000      73.800000      17.300000   27.900000  35.000000   
14           137.500000      70.000000      27.500000   92.000000  35.100000   
15           136.000000      70.000000      32.000000  110.000000  37.100000   
17           163.000000      72.000000      41.000000  114.000000  40.900000   

             DiabetesPedigreeFunction        Age   Outcome  
Pregnancies                                                 
0                            0.520838  27.603604  0.342342  
1                            0.486496  27.370370  0.214815  
2                            0.491660  27.194175  0.184466  
3                            0.432147  29.026667  0.360000  
4                            0.446353  32.779412  0.338235  
5                            0.396421  39.035088  0.368421  
6                            0.429520  39.340000  0.320000  
7                            0.443622  41.111111  0.555556  
8                            0.504711  45.368421  0.578947  
9                            0.550679  44.178571  0.642857  
10                           0.454167  42.666667  0.416667  
11                           0.522545  44.545455  0.636364  
12                           0.444333  47.444444  0.444444  
13                           0.463300  44.500000  0.500000  
14                           0.312000  42.000000  1.000000  
15                           0.153000  43.000000  1.000000  
17                           0.817000  47.000000  1.000000

##**Statistical Analysis**

Did we just see something about mean? There should be more like this. Total statistical analysis. This can be done in just a line!

In [68]:
dia_df.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

Now you might only want a few. Here's that:

In [69]:
dia_df.count()

Pregnancies                 768
Glucose                     768
BloodPressure               768
SkinThickness               768
Insulin                     768
BMI                         768
DiabetesPedigreeFunction    768
Age                         768
Outcome                     768
dtype: int64

In [70]:
dia_df.mean()

Pregnancies                   3.845052
Glucose                     120.894531
BloodPressure                69.105469
SkinThickness                20.536458
Insulin                      79.799479
BMI                          31.992578
DiabetesPedigreeFunction      0.471876
Age                          33.240885
Outcome                       0.348958
dtype: float64

In [71]:
dia_df.std()

Pregnancies                   3.369578
Glucose                      31.972618
BloodPressure                19.355807
SkinThickness                15.952218
Insulin                     115.244002
BMI                           7.884160
DiabetesPedigreeFunction      0.331329
Age                          11.760232
Outcome                       0.476951
dtype: float64

In [72]:
dia_df.min()

Pregnancies                  0.000
Glucose                      0.000
BloodPressure                0.000
SkinThickness                0.000
Insulin                      0.000
BMI                          0.000
DiabetesPedigreeFunction     0.078
Age                         21.000
Outcome                      0.000
dtype: float64

In [73]:
dia_df.max()

Pregnancies                  17.00
Glucose                     199.00
BloodPressure               122.00
SkinThickness                99.00
Insulin                     846.00
BMI                          67.10
DiabetesPedigreeFunction      2.42
Age                          81.00
Outcome                       1.00
dtype: float64

In [77]:
dia_df.quantile(0.75)

Pregnancies                   6.00000
Glucose                     140.25000
BloodPressure                80.00000
SkinThickness                32.00000
Insulin                     127.25000
BMI                          36.60000
DiabetesPedigreeFunction      0.62625
Age                          41.00000
Outcome                       1.00000
Name: 0.75, dtype: float64

##**Manipulating DataFrame**

Yes, here you go. But each function will show you the changed version, to save them in a dataset, you have to call a variable and assign them to it. Let's go!

In [79]:
dia_df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [80]:
dia_df.drop(index=0,axis=0)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
5              5      116             74              0        0  25.6   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
5                       0.201   30        0  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[767 rows x 9 columns]

You see axis=0, that is row removal. Simple. And vice versa. Index keyword is used to say row number.

In [81]:
dia_df.drop(index=7,axis=0)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[767 rows x 9 columns]

In [82]:
dia_df.drop(columns="BMI",axis=1)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin  \
0              6      148             72             35        0   
1              1       85             66             29        0   
2              8      183             64              0        0   
3              1       89             66             23       94   
4              0      137             40             35      168   
..           ...      ...            ...            ...      ...   
763           10      101             76             48      180   
764            2      122             70             27        0   
765            5      121             72             23      112   
766            1      126             60              0        0   
767            1       93             70             31        0   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 8 columns]

And yes, axis=1 is column removal!

In [83]:
dia_df.drop(columns="BloodPressure",axis=1)

Pregnancies  Glucose  SkinThickness  Insulin   BMI  \
0              6      148             35        0  33.6   
1              1       85             29        0  26.6   
2              8      183              0        0  23.3   
3              1       89             23       94  28.1   
4              0      137             35      168  43.1   
..           ...      ...            ...      ...   ...   
763           10      101             48      180  32.9   
764            2      122             27        0  36.8   
765            5      121             23      112  26.2   
766            1      126              0        0  30.1   
767            1       93             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 8 columns]

In [92]:
housing_df['Median House Value']=housing.target

Yes that was a column addition

In [93]:
housing_df.head()

0     1         2         3       4         5      6       7  Price  \
0  8.3252  41.0  6.984127  1.023810   322.0  2.555556  37.88 -122.23  4.526   
1  8.3014  21.0  6.238137  0.971880  2401.0  2.109842  37.86 -122.22  3.585   
2  7.2574  52.0  8.288136  1.073446   496.0  2.802260  37.85 -122.24  3.521   
3  5.6431  52.0  5.817352  1.073059   558.0  2.547945  37.85 -122.25  3.413   
4  3.8462  52.0  6.281853  1.081081   565.0  2.181467  37.85 -122.25  3.422   

   Median House Value  
0               4.526  
1               3.585  
2               3.521  
3               3.413  
4               3.422

# **Correlation**
There are two types of correlation. Positive and Negative. A function corr will show the correlation between all columns. Positive correleation between A and B means, if A increases, then B increases, too.

In [95]:
dia_df.corr()

Pregnancies   Glucose  BloodPressure  SkinThickness  \
Pregnancies                  1.000000  0.129459       0.141282      -0.081672   
Glucose                      0.129459  1.000000       0.152590       0.057328   
BloodPressure                0.141282  0.152590       1.000000       0.207371   
SkinThickness               -0.081672  0.057328       0.207371       1.000000   
Insulin                     -0.073535  0.331357       0.088933       0.436783   
BMI                          0.017683  0.221071       0.281805       0.392573   
DiabetesPedigreeFunction    -0.033523  0.137337       0.041265       0.183928   
Age                          0.544341  0.263514       0.239528      -0.113970   
Outcome                      0.221898  0.466581       0.065068       0.074752   

                           Insulin       BMI  DiabetesPedigreeFunction  \
Pregnancies              -0.073535  0.017683                 -0.033523   
Glucose                   0.331357  0.221071                  0.137337   
BloodPressure             0.088933  0.281805                  0.041265   
SkinThickness             0.436783  0.392573                  0.183928   
Insulin                   1.000000  0.197859                  0.185071   
BMI                       0.197859  1.000000                  0.140647   
DiabetesPedigreeFunction  0.185071  0.140647                  1.000000   
Age                      -0.042163  0.036242                  0.033561   
Outcome                   0.130548  0.292695                  0.173844   

                               Age   Outcome  
Pregnancies               0.544341  0.221898  
Glucose                   0.263514  0.466581  
BloodPressure             0.239528  0.065068  
SkinThickness            -0.113970  0.074752  
Insulin                  -0.042163  0.130548  
BMI                       0.036242  0.292695  
DiabetesPedigreeFunction  0.033561  0.173844  
Age                       1.000000  0.238356  
Outcome                   0.238356  1.000000

##**Done!**
That was a nice tour through Pandas, the rest can be learnt when specific need arises. These are the 90% used ones for Data Science.